entrypoints/internal.d.ts

```diff
@@ -0,0 +1,25 @@
+import type {StateChangeEvent} from '../types/state-change-events.d.cts';
+
+export type RunEvent = {
+	type: 'stateChange';
+	stateChange: StateChangeEvent;
+} | {
+	type: 'run';
+	plan: {
+		bailWithoutReporting: boolean;
+		debug: boolean;
+		failFastEnabled: boolean;
+		filePathPrefix: string;
+		files: string[];
+		matching: boolean;
+		previousFailures: number;
+		runOnlyExclusive: boolean;
+		firstRun: boolean;
+	};
+};
+
+export type {StateChangeEvent} from '../types/state-change-events.d.cts';
+
+export type Run = {
```

Comment:
open to a better name/export structure for this type

Reply:
`ObservedRun`?

Reply:
sure, I like that

Reply:
b7c80536d32ea4aea77c674d27ed8c46d7f9b63c